pkg/client/unversioned/testclient/fixture.go

```diff
@@ -60,6 +60,9 @@ func ObjectReaction(o ObjectRetriever, mapper meta.RESTMapper) ReactionFunc {
 
 	return func(action Action) (bool, runtime.Object, error) {
 		gvk, err := mapper.KindFor(action.GetResource())
+		if len(action.GetSubresource()) > 0 {
+			gvk, _ = mapper.KindFor(action.GetSubresource())
```

Comment:
ah because of deployment/scale. You don't need to scale the deployment down. What you need is:

1) GET deployment/name
2) GET rcs from deployment/name
3) STOP deployment/name
4) STOP rcs